sim-lib/src/cln.rs

```diff
@@ -101,10 +107,44 @@ impl LightningNode for ClnNode {
 
     async fn track_payment(
         &mut self,
-        _hash: PaymentHash,
-        _shutdown: Listener,
+        hash: PaymentHash,
+        shutdown: Listener,
     ) -> Result<PaymentResult, LightningError> {
-        unimplemented!()
+        loop {
+            tokio::select! {
+                biased;
+                _ = shutdown.clone() => {
+                    break Err(LightningError::TrackPaymentError("Shutdown before tracking results".to_string()));
+                },
+                response = self
+                .client
+                .list_pays(ListpaysRequest {
+                    payment_hash: Some(hash.0.to_vec()),
+                    ..Default::default()
+                }) => {
+                    let ListpaysResponse { pays } = response.map_err(|err| LightningError::TrackPaymentError(err.to_string()))?.into_inner();
+                    if let Some(pay) = pays.first() {
+                        let payment_status = ListpaysPaysStatus::from_i32(pay.status)
+                            .ok_or(LightningError::TrackPaymentError("Invalid payment status".to_string()))?;
+
+                        let payment_outcome = match payment_status {
+                            ListpaysPaysStatus::Pending => continue,
+                            ListpaysPaysStatus::Complete => PaymentOutcome::Success,
+                            // Task: https://github.com/bitcoin-dev-project/sim-ln/issues/26#issuecomment-1691780018
+                            ListpaysPaysStatus::Failed => PaymentOutcome::UnexpectedError,
+                        };
+
+                        break Ok(PaymentResult {
+                            // Task: https://github.com/bitcoin-dev-project/sim-ln/issues/26#issuecomment-1691780018
```

Comment:
I think this is the amount of attempts that LND required in order to complete the payment. CLN does not report that number on successful payments AFAIK.

I don't think we should care about this, and if we do, it should be an optional field in the outcome.